sql/src/main/java/io/crate/metadata/rule/ingest/IngestionService.java

```diff
@@ -87,13 +83,8 @@ public void clusterChanged(ClusterChangedEvent event) {
 
     private void filterOutInvalidRules(Map<String, Set<IngestRule>> ingestionRules) {
         for (Set<IngestRule> sourceIngestRules : ingestionRules.values()) {
-            Iterator<IngestRule> rulesIterator = sourceIngestRules.iterator();
-            while (rulesIterator.hasNext()) {
-                IngestRule ingestRule = rulesIterator.next();
-                if (schemas.tableExists(TableIdent.fromIndexName(ingestRule.getTargetTable())) == false) {
-                    rulesIterator.remove();
-                }
-            }
+            sourceIngestRules.removeIf(ingestRule ->
+                schemas.tableExists(TableIdent.fromIndexName(ingestRule.getTargetTable())) == false);
```

Comment:
Is this better? I didn't use a lambda here because it's capturing and will create a lambda factory object just to be able to link this call site to the lambda implementation.